WordPress/gutenberg
# Remove experimental flag from navigation and file blocks to use the Interactivity API (as private)

## What?
I opened a while ago a similar pull request to [remove the Interactivity API experimental flag](https://github.com/WordPress/gutenberg/pull/50941) completely. However, as the navigation and the file blocks have already been experimental for a while, and they are not adding new functionalities, I believe we can remove the experimental flag from them while keeping it for other blocks using the Interactivity API.

Keep in mind that it doesn't mean exposing the Interactivity API. It remains private only available for these blocks. This means that external blocks/plugins cannot use it, and we can keep experimenting and iterating.

## Why?
At this moment, users can only test the navigation and the file blocks using the Interactivity API if they activate the Experimental flag on the Gutenberg Experiments page.

## How?
These are the steps I took:

* I removed the conditional loading of the `script-loader.php` from the `load.php` file.
* I changed the `viewScript` field of the `block.json` of the Navigation and File blocks to read the interactivity.js file.
* I removed the old `view.js` files.
* I moved the directives directly to the `index.php` of the Image and File blocks.
* I added some logic to ensure the JS files are only loaded when they are necessary.
* I added some logic to load the directives conditionally.

Apart from that, I fixed a small bug where the `closeMenu` function was being triggered more than expected when pressing the ESC key when the menu was closed.

## Testing Instructions
* Test that the Navigation and the File blocks interactivity work as expected without the Experimental flag enabled.
* Test that the scripts are loaded conditionally:
    * In the navigation block: when the overlay is set to off, and the submenus options to "open on click" and "show arrow" are disabled, scripts and directives shouldn't be there.
    * In the file block: when the "Show inline embed" option is disabled, scripts and directives shouldn't be there.
